### PR TITLE
Handle files with MJD_OBS or MJD-OBS keywords (datastack module)

### PR DESCRIPTION
**Pull request by DougBurke**
Summary
=======

The `sherpa.astro.datastack` module has been updated to deal with files that use the `MJD-OBS` keyword instead of `MJD_OBS`. This is only used by the `show_stack` routine, which prints out
several keywords from each file in the stack (that is, it is not used when fitting the data, and is only in an "informational" routine).

Chandra data has historically used the `MJD_OBS` keyword, but will be switching to using `MJD-OBS`.

Details
=======

The test suite has been updated from SherpaTestCase to pytest style, and a basic test of `show_stack` has been added. A test suite for the `sherpa.astro.io.meta.Meta` class was added, along with a `__delitem__` method for the class (this was to make it easy to edit the header of the PHA files in memory rather than on disk).

The PR could be cleaned up to merge some of the initial commits, but it is ready for review now.

### CODE REVIEW

**Comment by DougBurke** [comment]
Interesting (perhaps) that the two test failures are with Python 3.7.

**Comment by DougBurke** [comment]
I can at least replicate the failure with Python 3.7

```
>       assert poly1.c0.frozen is False
E       AttributeError: 'NoneType' object has no attribute 'c0'
```

**Comment by DougBurke** [comment]
I've confirmed the test doesn't fail with python 3.5 or 3.6, just 3.7.

**Comment by DougBurke** [comment]
For some reason `sherpa.astro.datastack.ds._all_datastack_ids` isn't being cleared out in the python 3.7 case. My suspicion is that this is some difference in how the pytest "magic" is being handled, rather than anything intrinsic to the module (and could well have been introduced in my initial refactor from UnitTest style).

**Comment by DougBurke** [comment]
The code now looks for MJD-OBS, falling back to MJD_OBS if not found (which itself is optional). There's currently no test of the MJD-OBS handling.

**Comment by DougBurke** [comment]
I've now added a test of the MJD-OBS handling (which ended up requiring adding tests for the Meta class...)

**Comment by dtnguyen2** [comment]
When I saw @DougBurke recent sherpadev mail where he said:

> "...Do we have any documentation on what the Meta class is meant to do (it's not clear to me that it adds anything over a Python dictionary)?..."

I decided to test out his theory, so I commented out the current definition of the class Meta and wrote the following simple one:

```
class Meta(dict):
    """A key-value store."""

    def __init__(self):
        dict.__init__(self)
        return

    def pop(self, name):
        try:
            val = self[name]
        except KeyError:
            return None
        return val
```
So with the exception of the tests in the file _sherpa/astro/io/tests/test_meta.py_ all our current tests pass. So in theory: one can simplify things by redefine the ```Meta``` class (and remove the the _test_meta.py_ file).

**Comment by DougBurke** [comment]
Thanks for the investigation @dtnguyen2 - I don't think we should do anything about it now (with the 4.12 release coming up), but it's another thing we can put on our rather-extensive backburner!

File: sherpa/astro/datastack/ds.py [hunk 1]
```diff
@@ -118,22 +118,40 @@ def show_stack(self):
...
+                        timekey = key
+                        break
+                    except KeyError:
+                        pass
+
+            # This is informational, so do not necessarily need
+            # to report the actual key, but for now try to do so.
+            #
+            if timekey is None:
+                timekey = 'MJD-OBS'
```
Comment (wmclaugh) [inline sherpa/astro/datastack/ds.py]: in line 130, timekey is set to None and then changes to MJD-OBS if the keyword is not found. Would it be better to just initialize to MJD-OBS?

Comment (DougBurke) [reply sherpa/astro/datastack/ds.py]: I've made this change, and also tweaked the code to change from a hasattr check to try/except.

**Comment by DougBurke** [comment]
I don't know why it thinks the checks are still pending since  https://travis-ci.org/sherpa/sherpa/builds/592597602?utm_source=github_status&utm_medium=notification claims it finished (and passed) about 3 hours ago.

**Pull request merged**